### PR TITLE
Add a Scaleway dynamic inventory plugin

### PR DESCRIPTION
**Pull request by remyleone**
##### SUMMARY

Adds a Dynamic inventory plugin for Scaleway Compute

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

Scaleway Dynamic inventory

##### ANSIBLE VERSION

```
ansible 2.6.0 (scaleway_inventory e2833035ca) last updated 2018/04/13 16:26:35 (GMT +200)
  config file = None
  configured module search path = ['/Users/sieben/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/sieben/workspace/ansible/lib/ansible
  executable location = /Users/sieben/workspace/ansible/bin/ansible
  python version = 3.6.4 (v3.6.4:d48ecebad5, Dec 18 2017, 21:07:28) [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

### CODE REVIEW

**Comment by remyleone** [comment]
I don't understand what this UNSTABLE means

**Comment by bcoca** [comment]
@sieben shippable is trying to drive you crazy with random failures unrelated to your PR ... hence .. UNSTABLE

^ really means the test is not working properly, 99% of time due to external resource used being flakey

**Comment by remyleone** [comment]
@bcoca I've tried to rebase and push again but I get different errors all unrelated to my PR :/ What should we do?

**Comment by bcoca** [comment]
patience ... shippable has been hating every PR i look at lately....

**Comment by remyleone** [comment]
@bcoca How long should I wait for? Is there something I can do to speed it up?

File: lib/ansible/plugins/inventory/scaleway.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: scaleway
+    plugin_type: inventory
+    authors:
+      - Remy Leone <rleone@online.net>
+    short_description: Scaleway inventory source
+    description:
+        - Get inventory hosts from Scaleway
```
Comment (bcoca) [inline lib/ansible/plugins/inventory/scaleway.py:15]: you didn't document the 'regions' or 'tags' options

File: lib/ansible/plugins/inventory/scaleway.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        self.inventory.set_variable(server_id, "ipv4", server_info["public_ip"]["address"])
+
+    @property
+    def zones(self):
+        config_zones = self.config_data.get("regions", SCALEWAY_LOCATION.keys())
+        return set(SCALEWAY_LOCATION.keys()).intersection(config_zones)
+
+    @property
+    def tags(self):
+        return self.config_data.get("tags", None)
```
Comment (bcoca) [inline lib/ansible/plugins/inventory/scaleway.py]: See other plugins for how to use set_options/get_option base methods, these not only work from documentation but they also can add validation and defaults.

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/inventory/scaleway.py (hunk 1 above)
Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py:15]: done. Do you think it's clear enough?

File: lib/ansible/plugins/inventory/scaleway.py (hunk 2 above)
Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py]: I've renamed them to `_get_zones` and `_get_tags` is there additional steps I can do to improve my code?

File: lib/ansible/plugins/inventory/scaleway.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        - Get inventory hosts from Scaleway
+    options:
+        regions:
+            description: Filtrate results on a specific Scaleway region
+            type: list
+            default:
+                - ams1
+                - par1
+        tags:
+            description: Filtrate results on a specific tag
```
Comment (bcoca) [inline lib/ansible/plugins/inventory/scaleway.py]: filtrate is very liquid specific, use 'filter' instead

File: lib/ansible/plugins/inventory/scaleway.py (hunk 2 above)
Comment (bcoca) [reply lib/ansible/plugins/inventory/scaleway.py]: not what i meant, dont access config_data directly, using `get_option()` will verify type and ensure you get the default you set.

File: lib/ansible/plugins/inventory/scaleway.py (hunk 3 above)
Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py]: Done

File: lib/ansible/plugins/inventory/scaleway.py (hunk 2 above)
Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py]: Done tell me if it's ok now.

**Review by bcoca — APPROVED** [review]

File: lib/ansible/plugins/inventory/scaleway.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    return "/".join([api_endpoint, "servers"])
+
+
+class InventoryModule(BaseInventoryPlugin):
+    NAME = 'scaleway'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+        self.token = os.environ["SCW_TOKEN"]
```
Comment (s-hertel) [inline lib/ansible/plugins/inventory/scaleway.py]: It looks like scaleway modules allow a series of fallback environment variables as well as a module option. If you wanted to do the same thing for this plugin you could do
```
options:
    ...
    oauth_token:
        env:
            # in order of precedence
            - name: SCW_TOKEN
            - name: SCW_API_KEY
            - name: SCW_OAUTH_TOKEN
```
Then you can use self.get_option('oauth_token').

File: lib/ansible/plugins/inventory/scaleway.py (hunk 1 above)
Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py:15]: @bcoca Do you think the documentation is clear enough?

Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py:15]: Sorry you fixed the review. All good :)

File: lib/ansible/plugins/inventory/scaleway.py (hunk 4 above)
Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py]: Done. Do you have other reviews?

**Comment by blubber** [comment]
Guess this PR can be closed now that #38739 was merged?

**Pull request merged**